### ISSUE TITLE
libssl.so.1.1 missing 

### ISSUE THREAD

**Issue by butlepa1** [kneasle/ringing#300]
I just rebuilt my win11 machine / WSL / Ubuntu 22.04LTS. 

I pulled down the current build for momument but it failed with missing library.   libssl.slo.1.1

This seems to be known issue, you can fix it with

wget http://nz2.archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2.19_amd64.deb
sudo dpkg -i libssl1.1_1.1.1f-1ubuntu2.19_amd64.deb

Might be worth adding to the installation instructions.

**Comment by kneasle** [comment]
Huh, interesting - thanks for flagging!  If others are likely to run into this, it'd definitely be good to add it to the install instructions.

So, to clarify, this is a problem when downloading the latest release (from the GitHub releases page) rather than building from source (i.e. using the `cargo` command)?

**Comment by butlepa1** [comment]
Yes the issue came up with the downloading the latest release from GitHub.  However I  know in the past I built from source and I think it was the same problem as well that the library was missing.  Seems a common Ubuntu 22.04 issue.  I have not tested on Debian yet.  If i get a chance i will

Paul
________________________________

**Comment by kneasle** [comment]
OK, nice - I'd be surprised if a fresh build behaved differently since the release builds are built using `cargo`, just as when installing.

I've made a pull request to add install instructions to the guide.  Can you quickly check if what I've written is helpful before I merge it and make it public:
https://github.com/kneasle/ringing/blob/libssl-ubuntu/monument/cli/guide.md#library-libsslso11-not-found

Let me know any improvements and I'll fix and merge :)

**Comment by butlepa1** [comment]
Hi Ben

text looks fine

cheers

Paul

PS will you implementing Grandsire/Stedman features ?
________________________________

**Comment by kneasle** [comment]
> text looks fine 

Nice, I've merged the PR so it's now public.

> PS will you implementing Grandsire/Stedman features ?

So, this is a little tricky for two reasons.

Firstly, Grandsire singles will be tricky to implement because they affect _two_ changes (whereas every other call in common use only affects one).  Currently, Monument assumes that calls only overwrite one piece place notation, and it would require a lot of work to relax the assumption.

Secondly, Monument's goal isn't just to generate compositions, it wants to generate _human-friendly_ compositions.  For plain-bob-like methods, there aren't a lot of high-level structure conventions - enforcing tenors together with relatively few calls generally produces very human compositions.  But both Grandsire and Stedman have no such constraints, instead opting for mixing 'padding' courses (which keep some bells fixed to generate more of the same music) and 'turning' courses (which switch between these different types of music).  So, a human-friendly composition generator must be aware of this overall structure.

In summary, this may be possible (though very tricky) to achieve with Monument's current search algorithm, but to do this well you would really want a generator which designed specifically for Grandsire and/or Stedman.
